### PR TITLE
Add metrics for gossip/node sync process and SafeClient

### PR DESCRIPTION
**Pull request by longbowlu**
as title

### CODE REVIEW

File: crates/sui-core/src/authority.rs [hunk 1]
```diff
@@ -122,6 +122,7 @@ pub struct AuthorityMetrics {
     pub follower_connections: IntCounter,
     pub follower_connections_concurrent: IntGauge,
 
+    // TODO: consolidate these into GossipMetrics
```
Comment (gdanezis) [inline crates/sui-core/src/authority.rs:125]: Lets make an issue, tag it as "good first issue", for people joining the team and looking for ways to get into the codebase.

File: crates/sui-core/src/authority_active/gossip/mod.rs [hunk 2]
```diff
@@ -63,14 +67,79 @@ const REFRESH_FOLLOWER_PERIOD_SECS: u64 = 60;
 
 use super::ActiveAuthority;
 
+#[derive(Clone, Debug)]
+pub struct GossipMetrics {
+    pub concurrent_followed_validators: IntGauge,
```
Comment (gdanezis) [inline crates/sui-core/src/authority_active/gossip/mod.rs:72]: Maybe a comment over the structure that says: for description of each metric see the `new` function.

File: crates/sui-core/src/node_sync/node_follower.rs [hunk 3]
```diff
@@ -467,13 +473,16 @@ where
 
                     // wait until the tx becomes final before returning, so that the follower doesn't mark
                     // this tx as finished prematurely.
+                    let timer = self.metrics.wait_for_finality_latency_ms.start_timer();
```
Comment (gdanezis) [inline crates/sui-core/src/node_sync/node_follower.rs:476]: Oh, that is very cool -- had no idea that historgrams let you do this.

File: crates/sui-core/src/node_sync/node_follower.rs [hunk 4]
```diff
@@ -609,16 +619,20 @@ where
...
+                metrics.total_cert_download_attempts.inc();
+                let res = aggregator
                     .handle_transaction_and_effects_info_request(
                         digests,
                         authorities.as_ref(),
                         None,
                     )
-                    .await?
+                    .await?;
+                metrics.total_cert_downloads.inc();
```
Comment (gdanezis) [inline crates/sui-core/src/node_sync/node_follower.rs:634]: The names of these metrics might lead a metrics user to think that they represent actual downloads (with requests / responses on the wire and all that) whereas these are requests that could lead to one download, or a check we have the info. Lets either use a clearer name, or measure actual downloads?

File: crates/sui-core/src/safe_client.rs [hunk 5]
```diff
@@ -18,23 +20,102 @@ use sui_types::{
...
+        Self::new(&registry)
+    }
+}
+
 #[derive(Clone)]
 pub struct SafeClient<C> {
     authority_client: C,
     committee: Committee,
     address: AuthorityPublicKeyBytes,
+
```
Comment (gdanezis) [inline crates/sui-core/src/safe_client.rs:80]: Maybe a comment here to say the meaning of these can be found in the definition of `SafeClientMetrics::new` and that they are specialised metrics to this client / particular method.

**Review by gdanezis — APPROVED** [review]
Many thanks -- this LGTM. Have a look at the downloads metric that I think either needs renaming or its position changed (I think renaming is fine, as we measure downloads through safe client metrics)

File: crates/sui-core/src/authority.rs (hunk 1 above)
Comment (longbowlu) [reply crates/sui-core/src/authority.rs:125]: https://github.com/MystenLabs/sui/issues/3926

File: crates/sui-core/src/node_sync/node_follower.rs (hunk 4 above)
Comment (longbowlu) [reply crates/sui-core/src/node_sync/node_follower.rs:634]: the real metrics name is `gossip_total_cert_downloads` and I omit the prefix `gossip` to make it shorter.

I understand what you are saying but have a hard time coming up with a better name. I renamed them to `total_attempts_cert_downloads` and `gossip_total_attempts_cert_downloads` (so real metric name `gossip_total_attempts_cert_downloads` and `gossip_total_successful_attempts_cert_downloads`.

If they still don't make sense plz suggest one and I will put up another PR to udpate

**Pull request merged**